service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsControllerMidEventTest.java

```diff
@@ -0,0 +1,92 @@
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import static java.util.Objects.isNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(UploadC2DocumentsController.class)
+@OverrideAutoConfiguration(enabled = true)
+class UploadC2DocumentsControllerMidEventTest {
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String ERROR_MESSAGE = "A document must be uploaded";
+    private static final String DOCUMENT_URL = "http://localhost/documents/85d97996-22a5-40d7-882e-3a382c8ae1b4";
+    private static final String BINARY_URL = "http://localhost/documents/85d97996-22a5-40d7-882e-3a382c8ae1b4/binary";
+    private static final String FILENAME = "file.pdf";
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldReturnAnErrorWhenDocumentIsNotUploaded() throws Exception {
+        CallbackRequest request = createCallbackRequestWithTempC2Bundle(null,null,null);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldNotReturnAnErrorWhenDocumentIsUploaded() throws Exception {
+        CallbackRequest request = createCallbackRequestWithTempC2Bundle(DOCUMENT_URL,BINARY_URL,FILENAME);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
+    private CallbackRequest createCallbackRequestWithTempC2Bundle(String documentUrl,
+                                                                  String binaryUrl,
+                                                                  String filename) {
+        return CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .data(ImmutableMap.of(
+                    "temporaryC2Document",
+                    documentOrEmptyMap(documentUrl, binaryUrl, filename)))
+                .build())
+            .build();
+    }
+
+    private ImmutableMap documentOrEmptyMap(String documentUrl, String binaryUrl, String filename) {
+        if (isNull(documentUrl)) {
+            return ImmutableMap.of();
+        } else {
+            return ImmutableMap.of(
+                "document", ImmutableMap.of(
+                    "document_url", documentUrl,
+                    "document_binary_url", binaryUrl,
+                    "document_filename", filename));
+        }
```

Comment:
What do you think about following:
```
        return Optional.ofNullable(documentUrl)
            .map(url -> prepareDocumentMap(documentUrl, binaryUrl, filename))
            .orElse(ImmutableMap.of());
```
Not super neat as I don't like we are using the parameters from the method, but maybe something along the lines?

Reply:
removed this method entirely now based on discussion with you